src/Guzzle/SubjectRewritingMiddleware.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace eLife\Journal\Guzzle;
+
+use GuzzleHttp\Promise\PromiseInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use function GuzzleHttp\json_decode;
+use function GuzzleHttp\Psr7\parse_query;
+use function GuzzleHttp\Psr7\stream_for;
+
+final class SubjectRewritingMiddleware
+{
+    private $replacements = [
+        // Dummy rewrite
+        'cancer-biology' => [
+            'id' => 'immunology',
+            'name' => 'Immunology',
+        ],
+    ];
+
+    public function __invoke(callable $handler)
+    {
+        return function (RequestInterface $request, array $options) use (&$handler) {
+            $uri = $request->getUri();
+
+            foreach ($this->replacements as $replaced => $replacement) {
+                if (in_array($uri->getPath(), ["/highlights/{$replaced}", "/subjects/{$replaced}"])) {
```

Comment:
The highlights list (use on the subject page) will currently lose content, either the data source handles this, we splice the two together or we live with it.